lib/user.rb

```diff
@@ -0,0 +1,29 @@
+class User
+  attr_reader :name, :jokes, :id
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(new_joke)
+    @jokes << new_joke
+  end
+
+  def tell(audience, joke)
+    audience.jokes << joke
+  end
+
+  def joke_by_id(index)
```

Comment:
This works for the interaction pattern, only because the index is (unintentionally) related to the id.  This method would not work with an interaction pattern like this:

```
pry(main)> require './lib/joke'
# => true

pry(main)> require './lib/user'
# => true

pry(main)> sal = User.new("Sal")
# => #<User:0x00007fb71e1eb8d8...>

pry(main)> ali = User.new("Ali")
# => #<User:0x00007fb71e1a4348...>

pry(main)> joke_1 = Joke.new(43, "Why did the strawberry cross the road?", "Because his mother was in a jam.")    
# => #<Joke:0x00007fb71da169f0...>

pry(main)> joke_2 = Joke.new(13, "How do you keep a lion from charging?", "Take away its credit cards.")    
# => #<Joke:0x00007fb71d8e0bd0...>

pry(main)> sal.learn(joke_1)

pry(main)> sal.learn(joke_2)

pry(main)> sal.joke_by_id(43)
# => #<Joke:0x00007fb71da169f0...>

pry(main)> ali.joke_by_id(13)
# => #<Joke:0x00007fb71d8e0bd0...>
```

For the above interaction, you would need to do what you say in your comments: `need interate over array to return joke associated with @id from Joke`

You could accomplish this with `find`:

```
def joke_by_id(id)
  @jokes.find do |joke|
    joke.id == id
  end
end
```